peer/peer.go

```diff
@@ -243,6 +240,9 @@ type Config struct {
 	// Listeners houses callback functions to be invoked on receiving peer
 	// messages.
 	Listeners MessageListeners
+
+	// IdleTimeout is the duration of inactivity before we time out a peer.
+	IdleTimeout time.Duration
```

Comment:
what happens if IdleTimeout is unset?

Reply:
Agree, but what is still not clear to me is: what happens if any other member of Conf struct is unset.

I see that `newPeerBase` makes a copy of Conf.

Reply:
In go, every variable is zero/nil-initialized.

If I do not put IdleTimeout in the Config struct, it will be 0.  How does the peer package deal with that?